core/class.go

```diff
@@ -46,8 +57,270 @@ func (c *Cairo0Class) Version() uint64 {
+	var hintedClassHash *felt.Felt
+	var hintedClassHashErr error
+
+	externalEntryPointDigest := crypto.PedersenDigest{}
+	l1HandlerEntryPointDigest := crypto.PedersenDigest{}
+	constructorEntryPointDigest := crypto.PedersenDigest{}
+	builtInsDigest := crypto.PedersenDigest{}
+	dataDigest := crypto.PedersenDigest{}
+
+	wg.Add(6) //nolint:mnd
+
+	go func() {
+		defer wg.Done()
+		for _, ep := range definition.EntryPoints.External {
+			externalEntryPointDigest.Update(ep.Selector, ep.Offset)
+		}
+		externalEntryPointHash = externalEntryPointDigest.Finish()
+	}()
+
+	go func() {
+		defer wg.Done()
+		for _, ep := range definition.EntryPoints.L1Handler {
+			l1HandlerEntryPointDigest.Update(ep.Selector, ep.Offset)
+		}
+		l1HandlerEntryPointHash = l1HandlerEntryPointDigest.Finish()
+	}()
+
+	go func() {
+		defer wg.Done()
+		for _, ep := range definition.EntryPoints.Constructor {
+			constructorEntryPointDigest.Update(ep.Selector, ep.Offset)
+		}
+		constructorEntryPointHash = constructorEntryPointDigest.Finish()
+	}()
+
+	go func() {
+		defer wg.Done()
+		for _, builtIn := range program.Builtins {
+			builtInHex := hex.EncodeToString([]byte(builtIn))
+			builtInFelt, err := new(felt.Felt).SetString("0x" + builtInHex)
+			if err != nil {
+				return
+			}
+			builtInsDigest.Update(builtInFelt)
+		}
+		builtInsHash = builtInsDigest.Finish()
+	}()
+
+	go func() {
+		defer wg.Done()
+		hintedClassHash, hintedClassHashErr = computeHintedClassHash(definition.Abi, definition.Program)
+	}()
+
+	go func() {
+		defer wg.Done()
+		for _, data := range program.Data {
+			dataFelt, err := new(felt.Felt).SetString(data)
+			if err != nil {
+				return
+			}
+			dataDigest.Update(dataFelt)
+		}
+		dataHash = dataDigest.Finish()
+	}()
+
+	wg.Wait()
+
+	if hintedClassHashErr != nil {
+		return nil, hintedClassHashErr
+	}
+
+	clashHash := crypto.PedersenArray(
+		&felt.Zero,
+		externalEntryPointHash,
+		l1HandlerEntryPointHash,
+		constructorEntryPointHash,
+		builtInsHash,
+		hintedClassHash,
+		dataHash,
+	)
+
+	return clashHash, nil
+}
+
+// computeHintedClassHash calculates the hinted class hash by hashing the JSON combination
+// of ABI and Program.
+func computeHintedClassHash(abi, program json.RawMessage) (*felt.Felt, error) {
+	var mProgram Program
+	d := json.NewDecoder(bytes.NewReader(program))
+	d.UseNumber()
+	if err := d.Decode(&mProgram); err != nil {
+		return nil, err
+	}
+	err := mProgram.Format()
+	if err != nil {
+		return nil, err
+	}
+
+	formattedProgramBytes, err := json.Marshal(mProgram)
+	if err != nil {
+		return nil, err
+	}
+
+	formattedSpacesProgramStr, err := utils.ToPythonicJSON(string(formattedProgramBytes))
+	if err != nil {
+		return nil, err
+	}
+
+	stringifyABI, err := stringify(abi, nullSkipReplacer)
+	if err != nil {
+		return nil, err
+	}
+	formattedABI, err := utils.ToPythonicJSON(stringifyABI)
+	if err != nil {
+		return nil, err
+	}
+
+	// Combine both ABI and Program JSON strings
+	var hintedClassHashJSON strings.Builder
```

Comment:
I propose to replace it with `bytes.Buffer` it has almost the same API, but there is no need to cast to []byte from string at the bottom because we can just call `buffer.Bytes()`